Diagnostics: UI update.

mook-as opened the pull request:
Fixes #2913.

I'm using the grouping code that already exists in `<SortableTable>` to ensure we have a consistent style across RD & dashboard (especially since we bundle dashboard…)

Note: This will conflict with #2908 — it might be a good idea to delay reviewing until that one is done.

![diagnostics, opened](https://user-images.githubusercontent.com/3977982/190527655-0d1e2fcf-84d2-4bda-99df-2e6fb8af98f1.png)
![diagnostics, one closed](https://user-images.githubusercontent.com/3977982/190527674-e1005384-8c2a-4859-91ef-02124b449076.png)

For reference, grouped sortable table in the dashboard:
![grouped pods in dashboard](https://user-images.githubusercontent.com/3977982/190527789-d8940ea3-3c5d-499f-b247-2f0200ef6fd7.png)

jandubois commented:
> I'm using the grouping code that already exists in `<SortableTable>` to ensure we have a consistent style

Personally I find that the tab/label looks weird (especially when the heading gets longer), but I agree with the consistent style, so we should use whatever is already in place. And it nicely separates the "Mute" column heading without taking up extra space, so maybe it isn't so bad.

I also find that there is too much whitespace behind the end of the diagnostic text and the icon to link to a help topic. Can this be tightened up to be the same space you would have between 2 sentences?

Also this icon is misaligned, but maybe that is covered by #2903?

mook-as commented:
Yeah, the spacing between groups is a bit large, but _not_ having that makes it look weird too (maybe less of it would be useful?)  Either way, though, that should be done globally and not just for this instance.

I actually [added extra space](https://github.com/rancher-sandbox/rancher-desktop/pull/2916/files#diff-00be5f386b75c669db2d2cb2e4b90f45f28bb03e9dc8903dee8dd9f1d0cbee8dR265) because otherwise it was too close to the text; it's certainly possible to reduce the amount there to strike a better balance.

jandubois commented:
> let me know if this looks fine

The spacing between text and icon looks fine to me now, but the icon now seems to be too low.

I think it would look best if the bottom edge of the icon would sit exactly on top of the baseline of the text, just like another letter without descender. I guess it depends on if the icon has whitespace at the bottom, but otherwise `vertical-align: baseline` should do exactly that.

jandubois commented:
Also, if you look closely, then the "Mute" header is not centered on top of the switches:

<img width="192" alt="Screen Shot 2022-09-15 at 8 23 11 PM" src="https://user-images.githubusercontent.com/78947/190550150-55dc68bb-2105-4771-935d-79262398fc15.png">

jandubois commented:
Ok, 2 more picky comments:

The horizontal rule below the "Rerun" buttom was supposed to be removed.

And the left side margin in the box to the text is less than half of the right side margin between the switches and the box (I think they should be symmetrical):

<img width="898" alt="Screen Shot 2022-09-15 at 8 34 42 PM" src="https://user-images.githubusercontent.com/78947/190551354-a3d70073-4cc3-4209-98eb-01857e696419.png">

ericpromislow commented:
For some reason, the "Toggle Expand" button's td element is taking up a lot of space in the row, pushing everything else to the right.

Ran `npm run dev`, nothing special (I hope).

![Screen Shot 2022-09-16 at 2 31 11 PM](https://user-images.githubusercontent.com/108392/190788731-f817fdfc-b0ad-4439-8b26-d72bea1798b9.png)

mook-as commented:
The `Toggle Expand` expand button shouldn't exist anymore… fixes are behind a feature flag.

(But you're right, the colspan for the group heading is probably incorrect.)

jandubois commented:
> For some reason, the "Toggle Expand" button's td element is taking up a lot of space in the row,

There shouldn't be any expand/collapse functionality at the diagnostic level; it should only exist at the category level.

ericpromislow reviewed (changes requested):
Code looks fine, but I found a UI issue:

With only the `Testing:failed sample` visible:

* Ensure the Testing category is open (caret points down)
* Mute the sample test
* Hide muted tests - there should be no results now
* Unhide muted tests

The `Testing` caret is pointing down, but no content is visible. 
Click on the caret once -- it points sideways, but nothing changed.
Click again, so it points down -- content is now visible.

The content should be visible after the tests are unmuted.
*

ericpromislow commented on src/components/DiagnosticsBody.vue:
```diff
@@ -102,6 +107,24 @@ export default Vue.extend({
+      const groupRow = this.$refs[`group-${ group }`];
+
+      if (groupRow instanceof HTMLElement) {
+        groupRow.setAttribute('aria-expaneded', expanded ? 'true' : 'false');
```
Looks like a typo: 'aira-expaneded' -- should be `'aria-expaneded'` ?

ericpromislow replied:
Fixing this didn't fix the toggle problem I wrote up (but I didn't do a full restart).

mook-as commented:
Rebased, UI updated, and fixed the issue around collapse not restoring correctly (by making Vue handle more things).
![image](https://user-images.githubusercontent.com/3977982/191111043-d0ac24e7-502f-4079-8ec5-20dee02929a5.png)

ericpromislow commented:
No new screenshot needed. The second screenshot in https://github.com/rancher-sandbox/rancher-desktop/pull/2916#issue-1375220136 shows that the "Mute" label appears in a collapsed section. It shouldn't be displayed when that happens.

ericpromislow reviewed (changes requested):
I think the "Mute" label in the row's ".col-mute" element should be hidden when the row is collapsed. It's not doing anything in a collapsed row.

mook-as commented:
Adding a new screenshot anyway:

<img width="1052" alt="image" src="https://user-images.githubusercontent.com/3977982/191376344-e7bd79f2-9d6e-4620-bb54-624b3f939955.png">

(Ignore the line next to "Testing" in the collapsed column — I accidentally selected empty space.)

ericpromislow reviewed (approved):
Looks good now

ericpromislow reviewed (approved):
Survived rebase
